### PR TITLE
Blog app  add devise

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?

In this project, I will add the [devise](https://github.com/heartcombo/devise) gem to my app and use it for the registration and login of users.

Description of the task to be completed 👎

in this PR I will do bellow tasks 👎

- [x] Remove current_user method in ApplicationController, devise will provide us with one.
- [x] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).

- [x] Can register a new user.

- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A)

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (jerryowusu) [inline app/models/user.rb]: kindly ensure that the application checks for the confirmation of the email. You can do this by adding `:confirmable` to the code on lines 4 and 5.

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @mrkamin 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost down

## Highlights
Nice design :+1: 
Added devise gem :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (mrkamin) [reply app/models/user.rb]: @jerryowusu 
thanks for your good and valuable feedback,
now the application will check for confirmation.

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello Team! 👋 

### Things to Appreciate
- [ ] Implemented things requested by the previous Code Reviewer 👍 
- [ ] Professional README
- [ ] Descriptive PR summary
- [ ] Performed pair-programming 👌 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
@zainsadaqat 
thanks for your good and valuable feedback.

**Pull request merged**